sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/FileWriterFactory.scala

```diff
@@ -38,7 +38,7 @@ case class FileWriterFactory (
   @transient private lazy val jobId = SparkHadoopWriterUtils.createJobID(jobTrackerID, 0)
 
   override def createWriter(partitionId: Int, realTaskId: Long): DataWriter[InternalRow] = {
-    val taskAttemptContext = createTaskAttemptContext(partitionId)
+    val taskAttemptContext = createTaskAttemptContext(partitionId, realTaskId.toInt & Int.MaxValue)
```

Comment:
is it the same as `math.abs`?

Reply:
Yep, it same as `Math.abs(realTaskId.toInt)`

Reply:
https://github.com/apache/spark/blob/80addbbddab73980a0ea0672dac5634465c1997b/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/WriteFiles.scala#L97

https://github.com/apache/spark/blob/45ba9224602eb18fe45e339cbb8cf2e8a4924f0b/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FileFormatWriter.scala#L263

There are at least two other similar cases here, should we unify them as `math.abs`? Of course, this should be another PR. @cloud-fan

Reply:
I don't think perf matters here, and `math.abs` is definitely more readable.

Reply:
ok, we can unify the above three cases to `math.abs` in a follow-up.

Reply:
@cloud-fan If overflow `realTaskId.toInt & Int.MaxValue` and `math.abs` are not equal:

```scala
scala> val realTaskId = Long.MaxValue
val realTaskId: Long = 9223372036854775807

scala> val a = realTaskId.toInt
val a: Int = -1

scala> val b = realTaskId.toInt & Int.MaxValue
val b: Int = 2147483647

scala> val c= math.abs(realTaskId.toInt)
val c: Int = 1
```

Reply:
I see, `&` is better here.